crates/dojo-lang/src/plugin.rs

```diff
@@ -89,11 +90,14 @@ impl DojoPlugin {
 
 impl MacroPlugin for DojoPlugin {
     fn generate_code(&self, db: &dyn SyntaxGroup, item_ast: ast::Item) -> PluginResult {
+
         match item_ast {
             ast::Item::Module(module_ast) => self.handle_mod(db, module_ast),
             ast::Item::Struct(struct_ast) => {
                 let mut diagnostics = vec![];
-
+                
+                let indexed = is_indexed(db,  struct_ast.clone());
```

Comment:
we can move this into `component::handle_component_struct` since it gets the `struct_ast` as input

Reply:
yep makes sense